beacon-chain/types/block.go

```diff
@@ -142,34 +141,48 @@ func (b *Block) isSlotValid() bool {
 	return clock.Now().After(validTimeThreshold)
 }
 
-// IsValid is called to decide if an incoming p2p block can be processed.
-// It checks the slot against the system clock, and the validity of the included attestations.
-// Existence of the parent block and the PoW chain block is checked outside of this function because they require additional dependencies.
+// IsValid is called to decide if an incoming p2p block can be processed. It checks for following conditions:
+// 1.) Ensure parent processed.
+// 2.) Ensure pow_chain_ref processed.
+// 3.) Ensure local time is large enough to process this block's slot.
+// 4.) Attestation from proposer of block was included with the block in the network message.
```

Comment:
`Verify that the parent block's proposer's attestation is included` (I'm not sure how to phrase this without two consecutive possessive nouns lol)